pkg/workflow/compiler.go

```diff
@@ -1279,13 +1282,90 @@ func (c *Compiler) applyDefaultGitHubMCPAndClaudeTools(tools map[string]any) map
 		}
 	}
 
+	// Add Git commands and file editing tools when safe-outputs includes create-pull-request or push-to-branch
+	if safeOutputs != nil && needsGitCommands(safeOutputs) {
```

Comment:
This should moved into the ActionEngine interface to allow this kind of customization on other engines.